test/functional/p2p_tx_privacy.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.messages import (
+    msg_wtxidrelay,
+    msg_verack,
+    msg_tx,
+)
+from test_framework.p2p import (
+    P2PInterface,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.wallet import MiniWallet
+
+class P2PTxSpy(P2PInterface):
+    def on_version(self, message):
+        self.send_message(msg_wtxidrelay())
+
+class TxPrivacyTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def run_test(self):
+        self.wallet = MiniWallet(self.nodes[0])
+        self.wallet.rescan_utxos()
+
+        tx_originator = self.nodes[0].add_p2p_connection(P2PInterface())
+        spy = self.nodes[0].add_p2p_connection(P2PTxSpy(), wait_for_verack=False)
+        spy.wait_for_verack()
+
+        # tx_originator sends a tx
+        tx = self.wallet.create_self_transfer()["tx"]
+        tx_originator.send_and_ping(msg_tx(tx))
+
+        # Spy sends the verack
+        spy.send_and_ping(msg_verack())
+
+        # Spy should not get an inv for the tx
+        assert 'inv' not in spy.last_message
```

Comment:
Perhaps send a second tx from `tx_originator` to the node _after_ the version handshake with spy is complete, and then asset that the node announces the second tx, but not the first.